hub-templates/base-hub/templates/docs-service-deployment.yaml

```diff
@@ -0,0 +1,67 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: docs-service
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: docs-service
+  template:
+    metadata:
+      labels:
+        app: docs-service
+    spec:
+      volumes:
+        - name: nginx-config
+          configMap:
+            name: nginx-docs-configmap
+        - name: docs
+          emptyDir: {}
+      initContainers:
+        - name: docs-clone
+          image: alpine/git
+          args:
+            - clone
+            - --depth=1
+            - -b
+            - gh-pages
+            - --single-branch
+            - --
+            - {{ .Values.jupyterhub.docs.repo }}
+            - /srv/docs
+          securityContext:
+              runAsUser: 1000
+              allowPrivilegeEscalation: False
+              readOnlyRootFilesystem: True
+          volumeMounts:
+            - name: docs
+              mountPath: /srv/docs
+      containers:
+        - name: docs-sync
+          image: alpine/git
+          workingDir: /srv/docs
+          command: 
+            - /bin/sh
+          args:
+            - -c
+            - "while true; do git fetch origin; git reset --hard origin/master; sleep\
+              \ 5m; done"
+          securityContext:
+            runAsUser: 1000
+            allowPrivilegeEscalation: False
+            readOnlyRootFilesystem: True
+          volumeMounts:
+            - name: docs
+              mountPath: /srv/docs
+        - name: nginx-docs-service
+          image: nginx
```

Comment:
We should always specify a tag for all our images. This makes sure that our deployments are repeatable and reproducible. Without a tag, kubernetes tries to use the latest image. The 'latest image' can change over time without us knowing anything about it, and can break unexpectedly. For example, if the pod moves from one node to another for any reason, the new image pulled could be totally different from the old image! This could break our service when we aren't looking. So let's always find a tag and use it.